src/Nethermind/Nethermind.TxPool/Collections/SortedPool.cs

```diff
@@ -155,13 +157,17 @@ public EnhancedSortedSet<TValue> GetFirsts()
         /// <summary>
         /// Gets last element in supplied comparer order.
         /// </summary>
-        [MethodImpl(MethodImplOptions.Synchronized)]
-        public bool TryGetLast(out TValue last)
+        public bool TryGetLast(out TValue? last)
         {
-            last = _worstSortedValues.Max.Key;
+            KeyValuePair<TValue,TKey>? worstValue = _worstValue;
+            last = worstValue is not null ? worstValue.Value.Key : default;
```

Comment:
Still has same issue because `KeyValuePair<TValue,TKey>?` is a triple struct of ptr, ptr, bool; so could end up with a torn read; so the null check on `worstValue` is invalid as doesn't help with concurrently

Would have to do 
```
last = worstValue?.Value?.Key;
```

Reply:
```
public bool TryGetLast(out TValue? last)
{
    last = _worstValue.GetValueOrDefault().Key;
    return last is not null;
}
```